ci_framework/roles/hive/tasks/baremetal_deploy_cluster.yml

```diff
@@ -0,0 +1,99 @@
+    secret_name: "{{ cifmw_hive_baremetal_cluster_name }}-libvirtd-access"
+    secret: "{{ lookup('file', '{{ cifmw_hive_oc_provisioner_user_key.filename }}') }}"
+
+- name: Enable Hive access to the provisioner host
+  ansible.builtin.include_tasks:
+    file: oc_apply.yml
+  vars:
+    desc: "Provisioner SSH private key secret"
+    template_file: templates/private_key.yml.j2
+    dest_file: ocp_ssh_private_key.yml
+    secret_name: "{{ cifmw_hive_baremetal_cluster_name }}-provisioner-access"
+    secret: "{{ lookup('file', '{{ cifmw_hive_oc_provisioner_root_key.filename }}' ) }}"
+
+- name: Load baremetal install config
+  ansible.builtin.include_vars:
+    file: "{{ cifmw_hive_baremetal_install_config }}"
+    name: bm_install_config
+
+- name: Add dynamic values into the bm install config
+  ansible.builtin.set_fact:
+    bm_install_config: "{{ bm_install_config | combine(item, recursive=true) }}"
+  loop:
+    - {'metadata': {'name': "{{ cifmw_hive_baremetal_cluster_name }}" }}
+    - {'platform': {'baremetal': {'libvirtURI': "qemu+ssh://root@{{ cifmw_hive_oc_provisioner }}/system" }}}
+    - {'platform': {'baremetal': {'externalMACAaddress': "{{ cifmw_hive_oc_provisioner_ext_mac }}" }}}
+    - {'sshKey': "{{ cifmw_hive_baremetal_ocp_access_key }}" }
+
+- name: Gather the base domain of the OCP cluster
+  ansible.builtin.set_fact:
+    cifmw_hive_ocp_base_domain: "{{ bm_install_config.baseDomain }}"
+
+- name: Add baremetal install config secret
+  ansible.builtin.include_tasks:
+    file: oc_apply.yml
+  vars:
+    desc: "install config secret"
+    template_file: templates/bm_install_config.yml.j2
+    dest_file: bm_install_config.yml
+    secret_name: "{{ cifmw_hive_baremetal_cluster_name }}-install-config"
+    install_config_yml: "{{ bm_install_config | to_nice_yaml(indent=2) }}"
+
+- name: OpenShift cluster image resource
+  ansible.builtin.include_tasks:
+    file: oc_apply.yml
+  vars:
+    desc: "ocp cluster image"
+    template_file: templates/ocp_image.yml.j2
+    dest_file: ocp_image.yml
+    image_name: "{{ cifmw_hive_baremetal_cluster_name }}"
+    image_uri: "{{ cifmw_hive_baremetal_ocp_image }}"
+    kubeconfig: "{{ cifmw_hive_kubeconfig }}"
+
+- name: Initiate OCP deployment
+  ansible.builtin.include_tasks:
+    file: oc_apply.yml
+  vars:
+    desc: "OCP cluster deployment"
+    template_file: templates/bm_deploy_config.yml.j2
+    dest_file: bm_deploy_config.yml
+    known_host_entry: "{{ hostvars[cifmw_hive_oc_provisioner]['cifmw_hive_oc_provisioner_ssh_print'] | default(cifmw_hive_oc_provisioner_ssh_key) }}"
+
+- name: Wait until OCP cluster is deployed and configured
+  environment:
+    KUBECONFIG: "{{ cifmw_hive_kubeconfig }}"
+  ansible.builtin.command:
+    cmd: "oc get clusterdeployment {{ cifmw_hive_baremetal_cluster_name }} -o jsonpath='{ .spec.installed }' "
+  register: wait_result
+  until: wait_result.stdout | bool
+  retries: 360
+  delay: 30
+  when: not cifmw_hive_dry_run | bool
```

Comment:
care to move the `when` condition right bellow the `name` key? (not only for that task).
It makes reading far easier when we want to spot "what's running in what scenario". Thanks :)

Reply:
Done. Thank you for the suggestion.